src/components/App/App.js

```diff
@@ -24,11 +25,44 @@ type Props = {
   selectedProjectId: ?Project,
 };
 
-class App extends PureComponent<Props> {
+type State = {
+  isNodeJsMissing: boolean,
+};
+
+const { dialog, shell } = remote;
+
+class App extends PureComponent<Props, State> {
+  state = {
+    isNodeJsMissing: false,
+  };
+
   componentDidMount() {
-    window.addEventListener('beforeunload', this.killAllRunningProcesses);
+    const logAppLoaded = node_version => {
+      logger.logEvent('load-application', {
+        node_version: node_version || 'missing',
+      });
+    };
```

Comment:
We haven't really used this pattern much in the codebase, defining a function within an instance method.

What do you think about just inlining the logic?

```js
    checkIfNodeIsAvailable()
      .then(result => {
        logger.logEvent('load-application', { node_version: result.trim() });
      })
      .catch(error => {
        logger.logEvent('load-application', { node_version: 'missing' });
      })
      });
```

I know that it's less DRY, but I'm OK with that. I like that it gets a bit simpler, there's 2 distinct callsites with 2 distinct variables, instead of having to work out why there's an `||` in node_version.

Also, I wonder if we could pass `null` instead of a string, "missing". Do you know how Mixpanel handles this? Ultimately not a big deal though.

Reply:
Ah, I just saw in Mixpanel there is already a recorded `null`, from previous versions of the events.

I think we should just go with that? There is the issue of some "wrong" early data, but I feel like we will be totally ignoring data from before whenever 0.3 launches anyway :)

Reply:
OK, I just wanted to keep as DRY as possible. But two callsites are probably OK to duplicate code. I'll modify it. Also I'm OK with returning `null`. So the error case will be `logger.logEvent('load-application', { node_version: null})`.